src/reflect/scala/reflect/internal/Names.scala

```diff
@@ -470,6 +468,9 @@ trait Names extends api.Names {
     def isOperatorName: Boolean = decode != toString  // used by ide
     def longString: String      = nameKind + " " + decode
     def debugString = { val s = decode ; if (isTypeName) s + "!" else s }
+
+    override final def toString: String = if (cachedString == null) new String(chrs, index, len) else cachedString
```

Comment:
we could consider caching the name if we have calculated it. Its a trade off, and would depend on some benchmarking when in use

Reply:
We don't actually call `Name.toString` anywhere hot. When we build up full names (e.g. "scala.Boolean` or `scala/Boolean` we get the characters of the components from the underlying `chrs` array directly.

Reply:
I've added a commit to clean up the part of code I'm talking about here (`Symbol#fullName`). After that change, I've deprecated direct access to `Names.chrs`.

Reply:
if it is not useful, then why do we have the field at all?
There are lots of code paths that call toString, and probably don't all need to
e.g. append, prepend, string=

also the unapply of TermName and TypeName